src/ebonite/client/base.py

```diff
@@ -124,13 +123,112 @@ def build_service(self, name: str, model: 'core.Model', **kwargs) -> 'core.Image
             image = build_model_docker(name, model, **kwargs)
         return self.meta_repo.create_image(image)
 
-    @abstractmethod
-    def run_service(self, name: str, ports_mapping, detach: bool = True):
-        pass
+    def get_image(self, name: str, model: 'core.Model') -> 'core.Image':
+        """
+        Load image from repository
+
+        :param name: image name to load
+        :param model: :py:class:`~ebonite.core.objects.Model` instance to load image from
+        :return: loaded :py:class:`~ebonite.core.objects.Image` instance
+        """
+        return self.meta_repo.get_image_by_name(name, model)
+
+    def push_environment(self, environment: 'core.RuntimeEnvironment') -> 'core.RuntimeEnvironment':
+        """
+        Pushes runtime environment to repository
+
+        :param environment: environment to push
+        :return: same environment bound to repository
+        """
+        return self.meta_repo.create_environment(environment)
+
+    def get_environment(self, name: str) -> 'core.RuntimeEnvironment':
+        """
+        Load runtime environment from repository
+
+        :param name: name of environment to load
+        :return: loaded :py:class:`~ebonite.core.objects.RuntimeEnvironment` instance
+        """
+        return self.meta_repo.get_environment_by_name(name)
+
+    def run_service(self, name: str, image: 'core.Image', environment: 'core.RuntimeEnvironment' = None,
+                    **kwargs) -> 'core.RuntimeInstance':
+        """
+        Runs model service and stores it to repository
+
+        :param name: name of instance to run
+        :param image: image to run instance from
+        :param environment: environment to run instance in, if no given `localhost` is used
+        :return: :class:`~ebonite.core.objects.RuntimeInstance` instance representing run instance
+        """
+        env_name = 'localhost'
+        if environment is None:
+            environment = self.get_environment(env_name)
+        if environment is None:
+            environment = self.push_environment(core.RuntimeEnvironment(env_name))
+
+        params = {k: v for k, v in kwargs.items() if k in {'ports_mapping'}}
+
+        instance = core.RuntimeInstance(name, params=params)
+        instance.image = image
+        instance.environment = environment
+        instance = self.meta_repo.create_instance(instance)
+
+        from ebonite.build import run_docker_img, DockerImage
+        run_docker_img(name, DockerImage.from_core_image(image), environment.get_uri(),
```

Comment:
Here we hardcode `run_docker_img`, which is bad (I forgot if we discussed this already)

Reply:
Client itself doesn't expose Docker details. Currently we have no other implementations so I think for now it's fine.